src/pcm.c

```diff
@@ -1531,20 +1537,28 @@ static int pcm_mmap_transfer(struct pcm *pcm, void *buffer, unsigned int frames)
     }
 
     while (frames) {
+	pcm_sync_ptr(pcm, 0);
```

Comment:
I'm curious why do we need to call this. We have already synced the hw_ptr in `pcm_mmap_avail`.

Reply:
nice finding! Yes, there is no need to call pcm_sync_ptr(pcm, 0) here. 
pcm_sync_ptr(pcm, SNDRV_PCM_SYNC_PTR_HWSYNC); in pcm_mmap_avail() will not only update the ptr status but also sync with the hardware.
updated in the code.